src/cortxfs/cortxfs_ops.c

```diff
@@ -905,22 +905,71 @@ int cfs_rmdir(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 	return rc;
 }
 
+int cfs_unlink2(struct cfs_fh *parent_fh, struct cfs_fh *child_fh,
+                       cfs_cred_t *cred, char *name)
+{
+	int rc;
+
+	dassert(parent_fh && child_fh && cred && name);
+
```

Comment:
Since cfs_unlink2 is an api which would be exposed to upper layers, we need more validations other than NULL pointers for fh. I guess we can call fh invariants here to check if the user supplied fh is valid or not.

Reply:
Done addressed at 22e46940b661504ddfd7aeaec88f02b541cc9e56